src/OpenTelemetry.Instrumentation.AWSLambda/AWSLambdaWrapper.cs

```diff
@@ -20,13 +19,10 @@ namespace OpenTelemetry.Instrumentation.AWSLambda;
 /// </summary>
 public static class AWSLambdaWrapper
 {
-    private static readonly AssemblyName AssemblyName = typeof(AWSLambdaWrapper).Assembly.GetName();
+    internal const string ActivitySourceName = "OpenTelemetry.Instrumentation.AWSLambda";
 
-    [SuppressMessage("StyleCop.CSharp.OrderingRules", "SA1202:ElementsMustBeOrderedByAccess", Justification = "Initialization order.")]
-    internal static readonly string ActivitySourceName = AssemblyName.Name;
-
-    private static readonly Version Version = AssemblyName.Version;
-    private static readonly ActivitySource AWSLambdaActivitySource = new(ActivitySourceName, Version.ToString());
+    private static readonly string Version = typeof(AWSLambdaWrapper).Assembly.GetCustomAttribute<AssemblyFileVersionAttribute>()!.Version;
```

Comment:
Consider `AssemblyInformationalVersion.Split('+')[0]`.

eg. if you have versions 1.2.5-beta.1 this attribute should be equal to `1.2.5.beta.1+commit_hash`

Ref: https://github.com/adamralph/minver?tab=readme-ov-file#version-numbers and https://github.com/dotnet/sdk/issues/34568